Applications/DataExplorer/DataView/MshView.cpp

```diff
@@ -164,7 +165,17 @@ void MshView::openMap2dMeshDialog()
 	result->setName(dlg.getNewMeshName());
 	if (dlg.useRasterMapping())
 	{
-		if (!MeshLib::MeshLayerMapper::layerMapping(*result, dlg.getRasterPath(), dlg.getNoDataReplacement()))
+		std::unique_ptr<GeoLib::Raster> raster{
+		    FileIO::AsciiRasterInterface::readRaster(dlg.getRasterPath())};
+		if (!raster)
+		{
+			OGSError::box(QString::fromStdString(
+			    "Error mapping mesh. Could not read raster file " +
+			    dlg.getRasterPath()));
```

Comment:
This is a nice side effect: A possible error message can be displayed directly in the DataExplorer.